components/layout_2020/geom.rs

```diff
@@ -730,34 +743,50 @@ impl<T: Clone> Size<T> {
     }
 
     #[inline]
-    pub fn map<U>(&self, f: impl FnOnce(T) -> U) -> Size<U> {
+    pub(crate) fn map<U>(&self, f: impl FnOnce(T) -> U) -> Size<U> {
         match self {
             Size::Initial => Size::Initial,
             Size::MinContent => Size::MinContent,
             Size::MaxContent => Size::MaxContent,
             Size::FitContent => Size::FitContent,
+            Size::FitContentFunction(size) => Size::FitContentFunction(f(size.clone())),
             Size::Stretch => Size::Stretch,
             Size::Numeric(numeric) => Size::Numeric(f(numeric.clone())),
         }
     }
 
     #[inline]
-    pub fn maybe_map<U>(&self, f: impl FnOnce(T) -> Option<U>) -> Option<Size<U>> {
-        Some(match self {
-            Size::Numeric(numeric) => Size::Numeric(f(numeric.clone())?),
+    pub(crate) fn maybe_map_for_preferred<U>(&self, f: impl FnOnce(T) -> Option<U>) -> Size<U> {
+        match self {
+            Size::Numeric(numeric) => f(numeric.clone()).map_or(Size::Initial, Size::Numeric),
+            Size::FitContentFunction(numeric) => {
+                f(numeric.clone()).map_or(Size::FitContent, Size::FitContentFunction)
```

Comment:
Subjective nit: I personally find `.map_or(..)` confusing (which way round are the parameters) and would prefer `.map(...).unwrap_or(...)`. But that might just be me, I'm assuming that clippy pushes you this way.

Reply:
I wrote it this way because it's shorter, but there is a clippy lint indeed: https://rust-lang.github.io/rust-clippy/master/index.html#map_unwrap_or